tests/test_intero.py

```diff
@@ -0,0 +1,22 @@
+'''Unit tests for the intero.py module.'''
+
+import pytest
+
+
+@pytest.fixture
+def mock_vim(mocker):
+    mock = mocker.MagicMock()
+    mocker.patch.dict('sys.modules', vim=mock)
+    return mock
+
+
+def test_stack_dirname(mock_vim):
+    from intero import stack_dirname
+    mock_vim.configure_mock(**{'eval.return_value': '/home/foobar'})
+    assert stack_dirname() == '/home'
+
+
+def test_strip_internal(mock_vim):
+    from intero import strip_internal
+    assert strip_internal('\x1b[01;31mfoobar') == 'foobar'
+    assert strip_internal('\x1b[?2lfoobar') == 'foobar'
```

Comment:
It would be good to add a test case for the following prompt, since it uses 24-bit colour escape codes. 

     :set prompt "\ESC[38;2;255;100;0mλ> \ESC[m"

(That it happens to be *my* prompt is pure coincidence ;) )